### PR TITLE
Add uninstall function

### LINKED ISSUE #260: [Feature Request]: Uninstall Repository
Thanks to @mvdbeek , there is now a tool-removal API in galaxy: 

https://github.com/galaxyproject/galaxy/commit/67c66861c50f02475e497aa0c4feec7dfa16c2c2

As discussed in [this issue](https://github.com/galaxyproject/ephemeris/issues/83), it may make sense to add a bioblend tool uninstall endpoint, perhaps somewhere in this file:

https://github.com/galaxyproject/bioblend/blob/856db94ef09ad6cb026c2ed5b58c1a744cb9c641/bioblend/galaxy/toolshed/__init__.py

Maybe `uninstall_repository_revision` parallel to 

https://github.com/galaxyproject/bioblend/blob/856db94ef09ad6cb026c2ed5b58c1a744cb9c641/bioblend/galaxy/toolshed/__init__.py#L62-L68

### PR DESCRIPTION
**Pull request by erasche**
Fixes #260 ping @mvdbeek wanna add the ephemeris equivalent?

### CODE REVIEW

File: bioblend/galaxy/toolshed/__init__.py [hunk 1]
```diff
@@ -144,3 +144,36 @@ def install_repository_revision(self, tool_shed_url, name, owner,
...
+
+        :rtype: dict
+        :return: If successful, a dictionary with a message noting the removal
+        """
+
+        payload = {}
+        payload['tool_shed_url'] = tool_shed_url
+        payload['name'] = name
+        payload['owner'] = owner
+        payload['changeset_revision'] = changeset_revision
```
Comment (mvdbeek) [inline bioblend/galaxy/toolshed/__init__.py]: I think the changeset_revsion might be optional ? Can you add a test for this ?

**Comment by mvdbeek** [comment]
Sure can do.

File: bioblend/galaxy/toolshed/__init__.py (hunk 1 above)
Comment (erasche) [reply bioblend/galaxy/toolshed/__init__.py]: There wasn't a test for installation that I saw in the current tests. @nsoranzo do we want to add that? I can't do it this week but maybe next

File: bioblend/galaxy/toolshed/__init__.py [hunk 2]
```diff
@@ -144,3 +144,36 @@ def install_repository_revision(self, tool_shed_url, name, owner,
...
+        Uninstalls a specified repository revision from this Galaxy instance.
+
+        :type name: str
+        :param name: The name of the repository
+
+        :type owner: str
+        :param owner: The owner of the repository
+
+        :type changeset_revision: str
+        :param changeset_revision: The revision of the repository to be installed
```
Comment (nsoranzo) [inline bioblend/galaxy/toolshed/__init__.py]:
```suggestion
        :param changeset_revision: The revision of the repository to uninstall
```

File: bioblend/galaxy/toolshed/__init__.py (hunk 1 above)
Comment (nsoranzo) [reply bioblend/galaxy/toolshed/__init__.py]: If ephemeris is going to test for this, I guess it's fine with no tests here.

File: bioblend/galaxy/toolshed/__init__.py [hunk 3]
```diff
@@ -144,3 +144,36 @@ def install_repository_revision(self, tool_shed_url, name, owner,
...
+        """
+
+        payload = {}
+        payload['tool_shed_url'] = tool_shed_url
+        payload['name'] = name
+        payload['owner'] = owner
+        payload['changeset_revision'] = changeset_revision
+
+        url = "%s%s" % (self.gi.url, '/tool_shed_repositories')
+        print(url, payload)
```
Comment (nsoranzo) [inline bioblend/galaxy/toolshed/__init__.py]: Remove?
```suggestion
```

File: bioblend/galaxy/toolshed/__init__.py [hunk 4]
```diff
@@ -144,3 +144,36 @@ def install_repository_revision(self, tool_shed_url, name, owner,
...
+        :return: If successful, a dictionary with a message noting the removal
+        """
+
+        payload = {}
+        payload['tool_shed_url'] = tool_shed_url
+        payload['name'] = name
+        payload['owner'] = owner
+        payload['changeset_revision'] = changeset_revision
+
+        url = "%s%s" % (self.gi.url, '/tool_shed_repositories')
```
Comment (nsoranzo) [inline bioblend/galaxy/toolshed/__init__.py]:
```suggestion
```

File: bioblend/galaxy/toolshed/__init__.py [hunk 5]
```diff
@@ -144,3 +144,36 @@ def install_repository_revision(self, tool_shed_url, name, owner,
...
+        print(url, payload)
+
+        return self._delete(url=url, params=payload)
```
Comment (nsoranzo) [inline bioblend/galaxy/toolshed/__init__.py]:
```suggestion
        return self._delete(params=payload)
```

File: bioblend/galaxy/toolshed/__init__.py (hunk 1 above)
Comment (nsoranzo) [reply bioblend/galaxy/toolshed/__init__.py]: @mvdbeek I don't think `changeset_revision` is optional, see https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py#L643-L647

**Comment by nsoranzo** [comment]
@erasche @mvdbeek If you want to have a final look before I merge, I think it's ready.

**Comment by nsoranzo** [comment]
Thanks @erasche !

**Comment by erasche** [comment]
Thanks y'all!

**Pull request merged**